### ISSUE TITLE
Question about training and evaluation time steps

### ISSUE THREAD

**Issue by floatlazer** [pdebench/PDEBench#5]
Hi, thanks for sharing this great benchmark to facilitate the research! I have some questions when trying out your code.

#### Training steps
The last used time steps seem inconsistent in the "autoregressive" mode and the "single" mode. When training in the "autoregressive" mode, samples from `initial_step` to `t_train-1` are used (line 206 and 212):
https://github.com/pdebench/PDEBench/blob/99fbb7ae7ae33fc927ac1c3df8ee97e9c7601dd3/pdebench/models/fno/train.py#L204-L212
Note that the time step `t_train` is excluded in `range(initial_step, t_train)`, which I think should be correct. However, when training in the "single" mode, the prediction target is the ground truth sample at step `t_train`.
https://github.com/pdebench/PDEBench/blob/99fbb7ae7ae33fc927ac1c3df8ee97e9c7601dd3/pdebench/models/fno/train.py#L239-L241
I suppose the final time steps should be the same in these two modes, should it be `yy[..., t_train-1:t_train, :]` in the "single" mode?

#### Shallow water dataset 🐛 
Another question is about the sallow water dataset. In the paper, the total time step is 100, however, in the downloaded dataset there are 101 time steps. This will cause error in the FNO validation code where the prediction has 101 steps (line 269) while the label has 100 steps (line 282):
https://github.com/pdebench/PDEBench/blob/99fbb7ae7ae33fc927ac1c3df8ee97e9c7601dd3/pdebench/models/fno/train.py#L269-L271
https://github.com/pdebench/PDEBench/blob/99fbb7ae7ae33fc927ac1c3df8ee97e9c7601dd3/pdebench/models/fno/train.py#L282-L283
Should we use `range(initial_step, t_train)` in line 269?

#### Initial time steps in validation
Finally, when using "autoregressive" in validation, the initial time steps are also used to compute the error (line 283). To my understanding the initial steps should be the same as the ground truth and will have no error. Do we need to exclude these initial steps when computing the error?

Thank you!

**Comment by timothypraditia** [comment]
Hi @floatlazer, thank you very much for your input.
Here are our answers:

#### Training steps

The 'single' mode is only used for inverse problems. However, we also agree with your suggestion to make all the training mode consistent. Therefore, we have made the necessary changes in the code according to your suggestion.

#### Shallow water dataset

100 time steps that is written in the paper excludes the initial condition t=0. In total, there are 101 time steps in the data. The label will always have the same number of time steps as the dataset used, since the variable `t_train` only affects the training loop termination (and not slicing the loaded label itself). Additionally, we ensure that `t_train` does not exceed the length of the trajectory (see line 133-134 of the fno/train.py code) to prevent such errors. The reason we use `y.shape[2]` instead of `t_train`  in the validation is because we want to validate on the whole trajectory instead of chunks of the label. Sorry if it causes confusion.

#### Initial time steps in validation

We based our implementation on the original FNO code, and we did not realize this small detail. You are absolutely correct, and we have updated our code to exclude the initial time steps from the validation error calculation. Thank you very much for pointing this out!


I hope our answers clarify your questions.

**Comment by floatlazer** [comment]
Hi @timothypraditia,

Thanks a lot for you detailed replay. My questions are clarified, and with the new [commit](https://github.com/pdebench/PDEBench/commit/044a823d48d5d2d13fbc4928c6bb5e12ddff045e) the bug should be fixed.

I am closing the issue. Thanks again for sharing your great work!

**Comment by timothypraditia** [comment]
Hi there,

Thanks for noticing the indentation issue, I fixed it both for FNO and U-Net.

To answer your question, the validation is used only to implement early stopping, so it won't affect the reported results.

**Comment by floatlazer** [comment]
Got it, thank you!
